src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const currentState = { ...state };
+  const stateHistory = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      for (const date in obj.extraData) {
+        currentState[date] = obj.extraData[date];
+      }
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const date in obj.keysToRemove) {
+        for (const key in currentState) {
+          if (key === obj.keysToRemove[date]) {
+            delete currentState[key];
+          }
+        }
+      }
+    }
```

Comment:
Instead of nested loops for removing properties, you can use delete directly if the property exists.